### PR TITLE
Proper default value and documentation for $BASH_COMPLETIONS

### PR DESCRIPTION
**Pull request by blahgeek**
This is related to #1824

### CODE REVIEW

**Comment by scopatz** [comment]
LGTM @blahgeek! Waiting for CI. Thanks!

File: xonsh/platform.py [hunk 1]
```diff
@@ -305,16 +305,13 @@ def BASH_COMPLETIONS_DEFAULT():
...
-        else:
-            bcd = ('/usr/share/bash-completion', )
+        bcd = ('/usr/share/bash-completion/bash_completion', )
     elif ON_DARWIN:
         bcd = ('/usr/local/share/bash-completion/bash_completion',  # v2.x
                '/usr/local/etc/bash_completion')  # v1.x
     elif ON_WINDOWS and git_for_windows_path():
         bcd = (os.path.join(git_for_windows_path(),
-                            'usr\\share\\bash-completion'),
+                            'usr\\share\\bash-completion\\bash_completion'),
```
Comment (melund) [inline xonsh/platform.py]: Not sure that file exist. On my machine the directory looks like this: 
``` 
snail@sea C:\Program Fi\Git\usr\share $ tree /F /A bash-completion
Folder PATH listing
Volume serial number is 00000125 FA73:929B
C:\PROGRAM FILES\GIT\USR\SHARE\BASH-COMPLETION
|   python-argcomplete.sh
|
\---completions
        subversion
        svn
        svnadmin
        svndumpfilter
        svnlook
        svnsync
        svnversion
        vim
```

Comment (blahgeek) [inline xonsh/platform.py]: Those files under `..bash-completion/completions/` are installed along with its corresponding  packages (like vim), usually (as in linux) an additional package `bash-completion` is required to be installed which provides `..bash-completion/bash_completion` to lazy load those scripts. I don't know about windows, is there such a package?

Comment (astronouth7303) [inline xonsh/platform.py]: @blahgeek that's [Git for windows](https://git-for-windows.github.io/), which is a distribution of git and msys so windows users have a useful development environment.

I can confirm the file layout given by @melund for GFW.

Comment (melund) [inline xonsh/platform.py]: There doesn't seem to be a file called `bash-completions`. Maybe this is just leftovers from packaging the msys environment with git.  The git completion stuff is handled by this file `mingw64\\share\\git\\completion\\git-completion.bash`.

Are there any bash command that I could run that I could run to reveal how it is configured?

Comment (scopatz) [inline xonsh/platform.py]: Can we just include both paths in the default?  If it is not there, it should skip it right?

Comment (melund) [inline xonsh/platform.py]: Yes. It shouldn't matter. If the 'Git for Windows' project ever decide to include it it will start working.

Comment (blahgeek) [inline xonsh/platform.py]: Yes, and the original value `...\\bash-completion` is wrong anyway.

Comment (melund) [inline xonsh/platform.py]: That is true. Merging in now.

File: xonsh/completers/bash.py [hunk 2]
```diff
@@ -116,14 +116,9 @@ def complete_from_bash(prefix, line, begidx, endidx, ctx):
...
-def _collect_completions_sources():
-    sources = []
+def _get_completions_source():
     completers = builtins.__xonsh_env__.get('BASH_COMPLETIONS', ())
-    paths = (pathlib.Path(x) for x in completers)
-    for path in paths:
+    for path in map(pathlib.Path, completers):
         if path.is_file():
-            sources.append('source "{}"'.format(path.as_posix()))
-        elif path.is_dir():
```
Comment (mitnk) [inline xonsh/completers/bash.py]: @blahgeek I think this `elif` was mistakenly missed, and that's why bash completion now is not functional? :)

**Pull request merged**